File: 01-cpp-fundamentals/early-practice/hello.py
def calc_func(num1, num2, op):
    results = 0
    if op == 1:
        results = num1 + num2
    elif op == 2:
        results = num1 - num2
    elif op == 3:
        results = num1 * num2
    elif op == 4:
        results = num1 / num2
    return results

def display_menu():
    print("1. Addition")
    print("2. Subtraction")
    print("3. Multiplication")
    print("4. Division")
    print("5. Exit")
    print("6. Others")

def main():


    
    while True:
        operation = ""
        display_menu()
        op = int(input("Choose 1-5: "))
        if op == 1:
            operation = "+"
        elif op == 2:
            operation = "-"
        elif op == 3:
            operation = "*"
        elif op == 4:
            operation = "/"

            
        num1 = int(input("#1 number: "))
        num2 = int(input("#2 number: "))
        results = calc_func(num1, num2, op)
        print(f"The result is: {num1} {operation} {num2} = {results}")


main()

File: 01-cpp-fundamentals/early-practice/hello.cpp
#include <iostream>
using namespace std;

double calc_check(double num1, double num2, int op) {
    double result;

    switch (op) {
        case 1:
            result = num1 + num2;
            break;

        case 2:
            result = num1 - num2;
            break;

        case 3:
            result = num1 * num2;
            break;

        case 4:
            result = num1 / num2;
            break;

        default:
            return 0;
    }

    return result;
}

void display_menu() {
    cout << "1. Addition\n";
    cout << "2. Subtraction\n";
    cout << "3. Multiplication\n";
    cout << "4. Division\n";
    cout << "5. Exit\n";
}

int main() {
    int op;
    double num1, num2;
    double result;
    char operation;

    while (true) {
        cout << "========== CALCULATOR ==========" << endl;

        display_menu();

        cout << "Enter your choice (1-5): ";
if (!(cin >> op)) {
    cout << "Invalid input. Please enter a number from 1-5.\n";

    cin.clear();
    cin.ignore(10000, '\n');

    continue;
}

        if (op == 5) {
            cout << "Exiting the calculator. Goodbye!" << endl;
            break;
        }

        if (op < 1 || op > 5) {
            cout << "Invalid choice. Please try again." << endl;
            continue;
        }

        cout << "Enter #1: ";
        if (!(cin >> num1)) {
            cout << "Invalid input. Please enter a valid number.\n";
            cin.clear();
            cin.ignore(10000, '\n');
            continue;
        }

        cout << "Enter #2: ";
        if (!(cin >> num2)) {
            cout << "Invalid input. Please enter a valid number.\n";
            cin.clear();
            cin.ignore(10000, '\n');
            continue;
        }   

        result = calc_check(num1, num2, op);

        if (op == 1)
            operation = '+';
        else if (op == 2)
            operation = '-';
        else if (op == 3)
            operation = '*';
        else if (op == 4)
            operation = '/';

        cout << num1 << " " << operation << " "
            << num2 << " = " << result << endl;

        cout << "==================================" << endl;
    }

    return 0;
}